### PR TITLE
fix: border for merged cell

### PR DESCRIPTION
**Pull request by lumixraku**
close https://github.com/dream-num/univer-pro/issues/2561
close https://github.com/dream-num/univer/issues/3391

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by zhaolixin7** [comment]
1、水平、垂直合并后分别设置上、下、左、右后取消边框
表现：垂直的合并的地步边框未取消、水平的有边框未消失image

**Review by wzhudev — CHANGES_REQUESTED** [review]
There should be unit tests for the bugfix.

File: packages/sheets/src/commands/commands/set-border-command.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
 
+/* eslint-disable max-lines-per-function */
```
Comment (wzhudev) [inline packages/sheets/src/commands/commands/set-border-command.ts:17]: Do not add this. Please break long functions into smaller ones to make the code readable.

**Comment by zhaolixin7** [comment]
@lumixraku  单元格和合并单元格组合，设置内边框边框线预期：外边框线不应该显示
image

File: packages/sheets/src/commands/commands/set-border-command.ts (hunk 1 above)
Comment (lumixraku) [reply packages/sheets/src/commands/commands/set-border-command.ts]: image
break into multiple function

**Comment by lumixraku** [comment]
> There should be unit tests for the bugfix.

added

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**